client.go

```diff
@@ -26,6 +27,8 @@ type Conn struct {
 type Client struct {
 	Net            string            // if "tcp" a TCP query will be initiated, otherwise an UDP one (default is "" for UDP)
 	UDPSize        uint16            // minimum receive buffer for UDP messages
+	TLS            bool              // enables TLS connection (port 853)
```

Comment:
[ Quoting <notifications@github.com> in "Re: [dns] DNS over TLS (#300)..." ]
>> @@ -26,6 +27,8 @@ type Conn struct {
>>  type Client struct {
>>  	Net            string            // if "tcp" a TCP query will be initiated, otherwise an UDP one (default is "" for UDP)
>>  	UDPSize        uint16            // minimum receive buffer for UDP messages
>> +	TLS            bool              // enables TLS connection (port 853)
>
>@miekg I had a bit field in a new type in mind, but now that you mention it `Net`'s tempting—could that be reused?

But net allows "tcp6" and "udp4", should iwe support that as well?
So:

tcp6-tls
tcp4-tls
tcp-tls

for enabling this?

Seems rather clean to put this in the Net string.

/Miek

--
Miek Gieben